src/wp-includes/rest-api/endpoints/class-wp-rest-revisions-controller.php

```diff
@@ -402,6 +403,15 @@ public function get_item( $request ) {
 			return $revision;
 		}
 
+		if ( (int) $parent->ID !== (int) $revision->post_parent ) {
+			return new WP_Error(
+				'rest_post_invalid_parent',
+				/* translators: %d: A post revision id. */
+				sprintf( __( 'Parent post does not have a revision with id of "%d"' ), $request['id'] ),
```

Comment:
@ramonjd Should this message be overridden in `WP_REST_Template_Revisions_Controller`? 

https://github.com/WordPress/wordpress-develop/blob/c78e3fbce4c8af24b8feff47e6f40f9f7873a7db/src/wp-includes/rest-api/endpoints/class-wp-rest-template-revisions-controller.php#L169-L175

Similar messages there use the wording template.

Reply:
Good catch. I've used the same error code as the `WP_Error()` in the the `get_parent` method: `rest_post_invalid_parent`.

I'll update the mismatch error to something more unique and descriptive.

Reply:
@ramonjd I didn't mean the error code, I meant the message. It should say something like 
```
sprintf( __( 'This template does not have a revision with id of "%d"' ), $request['id'] ),
```

Reply:
Oh, apologies. Yeah, I guess "post" in the string isn't quite accurate in that context

`sprintf( __( 'Parent post does not have a revision with id of "%d"' ), $request['id'] )`

Rather than create more code in `class-wp-rest-template-revisions-controller.php`, what would you say about genericizing the message? E.g., 

`sprintf( __( 'Parent does not have a revision with id of "%d"' ), $request['id'] )`

I'll push a commit with that, but happy to extend `class-wp-rest-template-revisions-controller.php` as well.

Reply:
> This would mean either overriding the get_item method in template revisions controller or moving this logic to it's own method.

Now that I've had my coffee, I think I do prefer the messaging of `"'This template does not have a revision with id of "%d"'`.

Perhaps moving the mismatch logic to a single method might be more testable. I'll do that. 👍🏻

Reply:
The coffee has worn off, and I went with genericizing the error message further to avoid creating more code to maintain. 

I added a test to ensure the correct parent ID is returned in the error message.

```php
return new WP_Error(
    'rest_revision_parent_id_mismatch',
    /* translators: %d: A post id. */
    sprintf( __( 'The revision does not belong to the specified parent with id of "%d"' ), $parent->ID ),
    array( 'status' => 404 )
);
```